app/models/spree/user_decorator.rb

```diff
@@ -1,9 +1,13 @@
 Spree::User.class_eval do
 
+  INCOMPLETE_ORDER_STATES = [:cart, :address, :delivery, :payment, :confirm]
+
   delegate :firstname, :lastname, :full_address, :phone, to: :bill_address, allow_nil: true
 
-  scope :with_complete_orders,    -> { joins(:orders).merge(Spree::Order.complete).distinct }
-  scope :without_complete_orders, -> { where.not(id: with_complete_orders.pluck(:id)) }
+  scope :with_complete_orders,     -> { joins(:orders).merge(Spree::Order.complete).distinct }
+  scope :without_complete_orders,  -> { where.not(id: with_complete_orders.pluck(:id)) }
+  scope :with_incomplete_order,    -> { joins(:orders).where(spree_orders: { state: INCOMPLETE_ORDER_STATES }) }
```

Comment:
let's simply check for completed_at here.

Reply:
then it will include returned orders too. Do we want to consider them too?